File: Leetcode/sort_and_search/744_Find Smallest Letter Greater Than Target/Solution.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        int n = letters.size(), low = 0, high = n;
        while (low < high) {
            auto mid = (low + high) / 2;
            if (target < letters[mid]) high = mid;
            else low = mid + 1;
        }
        return letters[low%n];
    }
};

File: Leetcode/sort_and_search/744_Find Smallest Letter Greater Than Target/Solution.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        int length = letters.length, low = 0, high = length, mid = 0;
        while(low < high){
            mid = (low+high)/2;
            if(letters[mid]>target) high = mid;
            else low = mid+1;
        }
        return letters[low%length];
    }
}
